phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -1689,12 +1689,13 @@ public MutationState createIndex(CreateIndexStatement statement, byte[][] splits
                 for (ColumnName colName : requiredCols) {
                     // acquire the mutex using the global physical table name to
                     // prevent this column from being dropped while the index is being created
+                    String colNameSeparatedByDot = colName.getColumnName().replace(':', '.');
```

Comment:
Good to either extract this out as a SchemaUtil API and/or add a comment here. Is there some existing API which makes the indexed column name have a ':' instead of '.'?